src/MetadataViewer/MetadataViewerProvider.ts

```diff
@@ -0,0 +1,134 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as vscode from 'vscode';
+
+import {getMetadata} from './example/MetadataExample';
+import {MetadataViewer} from './MetadataViewer';
+
+export class RelationViewerDocument implements vscode.CustomDocument {
+  private readonly _uri: vscode.Uri;
+  private _metadataViwer: MetadataViewer[];
+
+
+  static async create(uri: vscode.Uri):
+      Promise<RelationViewerDocument|PromiseLike<RelationViewerDocument>> {
+    return new RelationViewerDocument(uri);
+  }
+
+  private constructor(uri: vscode.Uri) {
+    this._uri = uri;
+    this._metadataViwer = [];
+  }
+
+  public get uri() {
+    return this._uri;
+  }
+
+  // CustomDocument implements
+  dispose(): void {
+    // NOTE panel is closed before document and this is just for safety
+    this._metadataViwer.forEach((view) => {
+      while (this._metadataViwer.length) {
+        view.disposeMetadataView();
+      }
+    });
+    this._metadataViwer = [];
+  }
+
+  public openView(panel: vscode.WebviewPanel, extensionUri: vscode.Uri, fileUri: vscode.Uri) {
+    let view = new MetadataViewer(panel, extensionUri);
+
+    view.initWebView();
+    view.loadContent();
+    this._metadataViwer.push(view);
+
+
+    const seletedMetadata = getMetadata(fileUri);
+
+    // Sends a message to the web view with the metadata.
+    panel.webview.postMessage({command: 'showMetadata', metadata: seletedMetadata});
+
+    panel.onDidDispose(() => {
+      // TODO make faster
+      this._metadataViwer.forEach((view, index) => {
+        if (view.owner(panel)) {
+          view.disposeMetadataView();
+          this._metadataViwer.splice(index, 1);
+        }
+      });
+    });
+
+    return view;
+  }
+}
+
+export class MetadataViewerProvider implements
+    vscode.CustomReadonlyEditorProvider<RelationViewerDocument> {
+  public static readonly viewType = 'one.viewer.metadata';
+
+  private _context: vscode.ExtensionContext;
+
+  public static register(context: vscode.ExtensionContext): void {
+    const provider = new MetadataViewerProvider(context);
+
+    const registrations = [
+      vscode.window.registerCustomEditorProvider(MetadataViewerProvider.viewType, provider, {
+        webviewOptions: {
+          retainContextWhenHidden: true,
+        }
+      }),
+      vscode.commands.registerCommand(
+          'one.viewer.metadata.show',
+          async (uri) => {
+            // If the method is executed in the ONE Explorer, change the uri instance.
+            let fileUri = uri.uri ? uri.uri : uri;
```

Comment:
Could you separate 'one.viewer.metadata.show' command?

It does show up in both `context menu in default explorer` and `context menu in ONE Explorer`.

How about adding another command for `default vscode explorer`, like 'one.viewer.metadata.showInDefaultExplorer'?
Plus, if you do so, would you put the command title as 'ONE: Show Metadata', instead of 'Show Metadata' for the new command? Like, https://github.com/Samsung/ONE-vscode/pull/1315/files#r985182428

Reply:
ok!!, I agree this good suggestion. I`am going to apply this part right now.